server/templates.go

```diff
@@ -0,0 +1,248 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/bouk/httprouter"
+	"github.com/influxdata/chronograf"
+	"github.com/influxdata/chronograf/uuid"
+)
+
+// ValidTemplateRequest checks if the request sent to the server is the correct format.
+func ValidTemplateRequest(template *chronograf.Template) error {
+	switch template.Type {
+	default:
+		return fmt.Errorf("Unknown template type %s", template.Type)
+	case "query", "constant", "csv", "fieldKeys", "tagKeys", "tagValues", "measurements", "databases":
+	}
+
+	for _, v := range template.Values {
+		switch v.Type {
+		default:
```

Comment:
Yeah, that default on top is a style I've seen in the standard library to make it clear that it is an error condition first that matters most , while the remaining case statement entries are valid.  

It reads like, "return error unless"